### PR TITLE
DMP-1758: event-handler-for-sentencing-remarks

### PR DESCRIPTION
**Pull request by tonymort**
Adding SentencingRemarksAndRetentionPolicyHandler

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Test
+    void createsTranscriptionWithCorrectValues() {
+        dartsDatabase.createCourthouseUnlessExists(SWANSEA_COURTHOUSE);
+        var sentencingRemarksDartsEvent = createSentencingRemarksDartsEventFor(SWANSEA_COURTHOUSE);
+
+        eventDispatcher.receive(sentencingRemarksDartsEvent);
+
+        var persistedTranscriptions = dartsDatabase.getTranscriptionRepository().findAll();
+        var persistedTranscription = persistedTranscriptions.get(0);
```
Comment (mestebanez) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java:56]: May want to assert that there is a transcription before calling .get(0)

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Test
+    void createsTranscriptionCommentsWithCorrectValues() {
+        dartsDatabase.createCourthouseUnlessExists(SWANSEA_COURTHOUSE);
+        var sentencingRemarksDartsEvent = createSentencingRemarksDartsEventFor(SWANSEA_COURTHOUSE);
+
+        eventDispatcher.receive(sentencingRemarksDartsEvent);
+
+        var transcriptionComments = dartsDatabase.getTranscriptionCommentRepository().findAll();
+        assertThat(transcriptionComments.get(0).getComment()).isEqualTo("Transcription Automatically approved");
```
Comment (mestebanez) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java:88]: See above

File: src/main/java/uk/gov/hmcts/darts/event/service/handler/SentencingRemarksAndRetentionPolicyHandler.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                                                      ApplicationEventPublisher eventPublisher,
+                                                      TranscriptionsApi transcriptionsApi) {
+        super(retrieveCoreObjectService, eventRepository, hearingRepository, caseRepository, eventPublisher);
+        this.transcriptionsApi = transcriptionsApi;
+    }
+
+    @Override
+    @Transactional
+    public void handle(final DartsEvent dartsEvent, EventHandlerEntity eventHandler) {
+        var transcriptionRequestDetails = transcriptionRequestDetailsFrom(
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/event/service/handler/SentencingRemarksAndRetentionPolicyHandler.java:42]: The use of logging and logging levels within the code base is unclear to me. Do we need DEBUG/INFOs here

Comment (tonymort) [reply src/main/java/uk/gov/hmcts/darts/event/service/handler/SentencingRemarksAndRetentionPolicyHandler.java:42]: Added to event dispatcher

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java (hunk 1 above)
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java:57]: done

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java (hunk 2 above)
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/SentencingRemarksAndRetentionPolicyHandlerTest.java:90]: done

**Review by mestebanez — APPROVED** [review]

**Pull request merged**